test/unit/MerkleDrop.spec.js

```diff
@@ -219,9 +237,116 @@ describe(">> MerkleDrop", () => {
       });
+
+      it("reverts if tranch is expired", async () => {
+        await merkleDropInstance.connect(prime).expireTranche(trancheIdx);
+        expect(
+          merkleDropInstance
+            .connect(alice)
+            .claimTranche(alice.address, trancheIdx, expectedBalance, proof)
+        ).to.be.revertedWith("Incorrect merkle proof");
+      });
     });
 
-    describe("$ allocation is zero", () => {
+    describe("$ with second tranche", async () => {
+      let secondMerkleRoot,
+        secondTree,
+        aliceFirstProof,
+        aliceFirstExpectedBalance;
+
+      const firstTrancheIdx = "0";
+      const secondTrancheIdx = "1";
+      const aliceSecondClaim = "10";
+      const bobSecondClaim = "20";
+      const initialState = {
+        ...commonState,
+        withProof: true,
+      };
+
+      beforeEach("add a second tranche", async () => {
+        ({
+          proof: aliceFirstProof,
+          expectedBalance: aliceFirstExpectedBalance,
+        } = await setupInitialState(contractInstances, initialState));
+
+        const secondAllocation = [
+          [alice.address, aliceSecondClaim],
+          [bob.address, bobSecondClaim],
+        ];
+        const secondCumulativeAllocation = utils
+          .parseEther(aliceSecondClaim)
+          .add(utils.parseEther(bobSecondClaim));
+        const secondTranche = getTranche(...secondAllocation);
+
+        secondTree = createTreeWithAccounts(secondTranche);
+        secondMerkleRoot = secondTree.hexRoot;
+
+        await v2TokenInstance
+          .connect(root)
+          .increaseAllowance(
+            merkleDropInstance.address,
+            secondCumulativeAllocation
+          );
+        await merkleDropInstance
+          .connect(root)
+          .seedNewAllocations(secondMerkleRoot, secondCumulativeAllocation);
+      });
+
+      it("lets alice claim her first and second claim", async () => {
+        await merkleDropInstance
+          .connect(alice)
+          .claimTranche(
+            alice.address,
+            firstTrancheIdx,
+            aliceFirstExpectedBalance,
+            aliceFirstProof
+          );
+        const aliceSecondProof = getAccountBalanceProof(
+          secondTree,
+          alice.address,
+          utils.parseEther(aliceSecondClaim)
+        );
+        await merkleDropInstance
+          .connect(alice)
+          .claimTranche(
+            alice.address,
+            secondTrancheIdx,
+            utils.parseEther(aliceSecondClaim),
+            aliceSecondProof
+          );
+
+        const expectedTotalBalance = utils
+          .parseEther(aliceSecondClaim)
+          .add(aliceFirstExpectedBalance);
+        expect(await v2TokenInstance.balanceOf(alice.address)).to.eq(
+          expectedTotalBalance
+        );
+      });
+
+      describe("$ with first tranche expired", () => {
+        it("lets alice claim her second claim", async () => {
```

Comment:
Is is correct that we cannot claim the first tranche after we claimed the second one? Or we won't be able to pick the first one only after it was expired by function call?

Reply:
As a user, we can claim basically from any tranche that
- is not expired
- has not yet been claimed by the same user

I added another test case to make this clear.